Sources/MacOSPlatform/MacOS.swift

```diff
@@ -0,0 +1,184 @@
+            let executableURL = toolchainBinURL.appendingPathComponent(executable)
+
+            // Deletion confirmed with user above.
+            try linkURL.deleteIfExists()
+
+            try FileManager.default.createSymbolicLink(
+                atPath: linkURL.path,
+                withDestinationPath: executableURL.path
+            )
+        }
+
+        SwiftlyCore.print("""
+            NOTE: On macOS it is possible that the shell will pick up the system Swift on the path
+            instead of the one that swiftly has installed for you. You can run the 'hash -r'
+            command to update the shell with the latest PATHs.
+
+                hash -r
+
+            """
+        )
+
+        return true
+    }
+
+    public func unUse(currentToolchain: ToolchainVersion) throws {
+        let currentToolchainBinURL = self.swiftlyToolchainsDir
+            .appendingPathComponent(currentToolchain.identifier + ".xctoolchain", isDirectory: true)
+            .appendingPathComponent("usr", isDirectory: true)
+            .appendingPathComponent("bin", isDirectory: true)
+
+        for existingExecutable in try FileManager.default.contentsOfDirectory(atPath: currentToolchainBinURL.path) {
+            guard existingExecutable != "swiftly" else {
+                continue
+            }
+
+            let url = self.swiftlyBinDir.appendingPathComponent(existingExecutable)
+            let vals = try url.resourceValues(forKeys: [URLResourceKey.isSymbolicLinkKey])
+
+            guard let islink = vals.isSymbolicLink, islink else {
+                throw Error(message: "Found executable not managed by swiftly in SWIFTLY_BIN_DIR: \(url.path)")
+            }
+            let symlinkDest = url.resolvingSymlinksInPath()
+            guard symlinkDest.deletingLastPathComponent() == currentToolchainBinURL else {
+                throw Error(message: "Found symlink that points to non-swiftly managed executable: \(symlinkDest.path)")
+            }
+
+            try self.swiftlyBinDir.appendingPathComponent(existingExecutable).deleteIfExists()
+        }
+    }
+
+    public func listAvailableSnapshots(version _: String?) async -> [Snapshot] {
+        []
+    }
+
+    public func getExecutableName(forArch: String) -> String {
+        "swiftly-\(forArch)-macos-osx"
+    }
+
+    public func currentToolchain() throws -> ToolchainVersion? { nil }
+
+    public func getTempFilePath() -> URL {
+        FileManager.default.temporaryDirectory.appendingPathComponent("swiftly-\(UUID()).pkg")
+    }
+
+    public func verifySignature(httpClient: SwiftlyHTTPClient, archiveDownloadURL: URL, archive: URL) async throws {
+        // No signature verification is required on macOS since the pkg files have their own signing
+        //  mechanism and the swift.org downloadables are trusted by stock macOS installations.
+    }
+
+    private func runProgram(_ args: String..., quiet: Bool = false) throws {
```

Comment:
I'm a little hesitant to do that since I expect that the `/usr/bin/env` path trick might not work with Windows, so only 2/3 platforms can use this as-is. I know that duplication can be painful at times, but so can leaky abstractions.

Reply:
Yeah, I agree that once we start to introduce Windows support, we won't want it as a default implementation, but until then, I think it'd be best to cut down on the duplication though. As an alternative, this logic could be factored out into a separate `unixRunProgram` function that both implementations call. Either way is fine with me.

Reply:
I've added a Linux/macOS only runProgram to the protocol, guarded with `#if` in case another platform, like Windows, tries to use it.